src/index.js

```diff
@@ -35,37 +38,55 @@ export default async function educandu(options) {
     process.exit(hasError ? 1 : 0);
   });
 
-  process.on('uncaughtException', err => {
-    logger.fatal(err);
-    Graceful.exit(1);
-  });
-
   try {
-    logger.info('Starting application');
-
     container = await bootstrapper.createContainer(options);
+    const maintenanceService = container.get(MaintenanceService);
     const educanduServer = container.get(EducanduServer);
     const serverConfig = container.get(ServerConfig);
+    const userService = container.get(UserService);
 
-    logger.info('Starting server');
-    educanduServer.listen((err, port) => {
-      if (err) {
-        logger.fatal(err);
-        Graceful.exit(1);
+    logger.info('Starting application');
+
+    const runMaintenance = !serverConfig.skipMaintenance;
+
+    logger.info(`Starting server${runMaintenance ? ' in maintenance mode' : ''}`);
+    const port = await educanduServer.listen({ maintenance: runMaintenance });
+    logger.info(`Server listening on http://localhost:${port}`);
```

Comment:
Maybe we can run this at the end, after `Application started successfully`. I know it's already listening but this info is lost halfway in the logs 😬

Reply:
Done